pkg/events/trigger/context.go

```diff
@@ -1,72 +1,83 @@
 package trigger
 
 import (
+	"fmt"
 	"sync"
 
 	"github.com/aquasecurity/tracee/pkg/counter"
 	"github.com/aquasecurity/tracee/pkg/errfmt"
 	"github.com/aquasecurity/tracee/pkg/events/parse"
+	"github.com/aquasecurity/tracee/pkg/logger"
 	"github.com/aquasecurity/tracee/types/trace"
 )
 
 const ContextArgName = "caller_context_id"
 
+// Context is an interface for a struct used to store the triggering events context.
 type Context interface {
 	Store(trace.Event) uint64               // store an invoke context
-	Get(uint64) (trace.Event, bool)         // get an invoked event context
+	Load(uint64) (trace.Event, bool)        // loads an invoked event context
 	Apply(trace.Event) (trace.Event, error) // apply an invoked event context (implicitly gets the event)
 }
 
-// context is a struct used to store the triggering events context
 type context struct {
 	store   map[uint64]trace.Event
-	mutex   sync.RWMutex
+	mutex   *sync.Mutex
```

Comment:
I guess this is ok because we don't have many frequent user triggered events... But when we do couldn't this be a pretty contentious lock?

Reply:
The Store() involves copying trace.Event (needs locking, can't be atomic), the Load involves another copy (also needs locking). We could return a pointer but then the map deletion from the Load() would also require locking. No free lunch :\. We can improve this in the future if needed. I think we should first make sure all locks work correctly and then profile lock contention to optimize those.

Reply:
Ok I see that we really only used the read lock for one happy path anyway... Resolving.